src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/Native/SafeHandles.cs

```diff
@@ -12,22 +13,19 @@ namespace Internal.Cryptography.Pal.Native
     /// </summary>
     internal abstract class SafePointerHandle<T> : SafeHandle where T : SafeHandle, new()
     {
-        public SafePointerHandle()
+        protected SafePointerHandle()
             : base(IntPtr.Zero, true)
         {
         }
 
-        public sealed override bool IsInvalid
+        internal static T CreateInvalid()
         {
-            get
-            {
-                return handle == IntPtr.Zero;
-            }
+            return new T();
```

Comment:
In theory I like the suggestion.  In practice it makes me a bit nervous.  Since CreateInvalid is just another name for the constructor, a derived type could easily do something like:
```C#
public SafeSomethingHandle Create(IntPtr ptr)
{
    var h = CreateInvalid();
    h.SetHandle(ptr);
    return h;
}
```
and now we run the risk of never free'ing the handle.  Since the API does make it possible to reassign the handle values, I'm hesitant to suppress finalization anywhere else than in the API  generally understood to close the handle (at least in a base type like this).